core/src/impl/Kokkos_ViewMapping.hpp

```diff
@@ -2887,12 +2887,22 @@ struct ViewValueFunctor<DeviceType, ValueType, false /* is_scalar */> {
   construct_dispatch() {
     ValueType value{};
     if (Impl::is_zero_byte(value)) {
+      uint64_t kpID = 0;
+      if (Kokkos::Profiling::profileLibraryLoaded()) {
+        Kokkos::Profiling::beginParallelFor(
+            "Kokkos::View::initialization [" + name + "]", 0, &kpID);
```

Comment:
So, I'm not sure whether "0" is the correct deviceID anymore, but that's fine for this PR. We should make a note that it's on me to fix the places where we used the "just say 0, lol" methodology for assigning deviceID's

Reply:
Perhaps this should be described as happening on

```c++
space.impl_instance_id();
```

Yeah, that's probably the most fair way to say what's happening

Reply:
Also, sorry I didn't catch that one last time around, hate giving you the runaround on this stuff

Reply:
I addressed your suggestions now and I am using `space.impl_instance_id()` now. Note that `negin_parallel_for` is using `Kokkos::Profiling::Experimental::device_id(policy.space())` instead. Let me know if you think that this is fine, if you want something else or if I should just add a comment for you to revisit.

Reply:
Mirror what's done in the other place, I can see why I phrased it that way